app/_plugins/generators/redirects.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+module Jekyll
+  class Redirects < Jekyll::Generator
+    priority :low
+
+    def generate(site)
+      current_redirects = File.read("app/_redirects")
+      active_versions = site.data['versions'].filter {|v| v['release'] != "dev"}
+
+      # Generate redirects for the latest version
+      latest_release = active_versions.last['release']
+      redirects = [
+        "# Generated redirects",
+        "/docs /docs/#{latest_release} 301",
+        "/install/kong-gateway /docs/#{latest_release}/explore/gateway 301",
+        "/docs/latest/documentation/gateway /docs/#{latest_release}/explore/gateway 301",
+        "/docs/latest/deployments /docs/:version/#{latest_release}/deployments 301",
+        "/docs/latest/documentation/deployments /docs/#{latest_release}/introduction/deployments 301",
+        "/docs/latest/explore/backends /docs/#{latest_release}/documentation/configuration 301",
+        "/install /install/#{latest_release} 301",
+        "/docs/latest/* /docs/#{latest_release}/:splat 301",
+        "/install/latest/* /install/#{latest_release}/:splat 301",
+        "/docs/:version/policies/ /docs/:version/policies/introduction 301",
+        "/docs/:version/overview/ /docs/:version/overview/what-is-kuma 301",
+        "/docs/:version/other/ /docs/:version/other/enterprise 301",
+        "/docs/:version/installation/ /docs/:version/installation/kubernetes 301",
+        "/docs/:version/api/ /docs/:version/documentation/http-api 301",
+        "/docs/:version/documentation/deployments/ /docs/:version/introduction/deployments 301",
+        "/latest_version.html /latest_version 301",
+      ].join("\n")
+
+      # Generate redirects for specific versions
+      redirects = "#{redirects}\n\n# Specific version to release\n"
+      active_versions.each do |v|
+        vp = v['version'].split('.').map(&:to_i)
+        idx = 0
+        loop do
```

Comment:
nit: isn't `for idx in 0.. vp[2] do` a little easier to follow than the loop with a break?

In any case I think a comment explaining that you are generating redirects for all patch versions from `x.y.0` until latest version maybe useful for future readers